File: eBiblioteka.Servisi/Database/KorisnikIzabranaKnjiga.cs
using System;
using System.Collections.Generic;

namespace eBiblioteka.Servisi.Database;

public partial class KorisnikIzabranaKnjiga
{
    public int KorisnikIzabranaKnjigaId { get; set; }

    public int KnjigaId { get; set; }

    public int KorisnikId { get; set; }

    public bool IsChecked { get; set; }

    public DateTime DatumRezervacije { get; set; }

    public DateTime DatumVracanja { get; set; }

    public virtual Knjiga? Knjiga { get; set; }

    public virtual Korisnik? Korisnik { get; set; }
}

File: eBibliotekaUI/ebiblioteka_mobile/lib/models/korisnik_izabrana_knjiga.dart
import 'package:ebiblioteka_mobile/models/knjiga.dart';
import 'package:json_annotation/json_annotation.dart';

part 'korisnik_izabrana_knjiga.g.dart';

@JsonSerializable()
class KorisnikIzabranaKnjiga {
  int? korisnikIzabranaKnjigaId;
  int? korisnikId;
  int? knjigaId;
  DateTime? datumRezervacije;
  DateTime? datumVracanja;
  bool? isChecked;
  Knjiga? knjiga;

  KorisnikIzabranaKnjiga(
      {this.korisnikIzabranaKnjigaId,
      this.korisnikId,
      this.knjigaId,
      this.datumRezervacije,
      this.datumVracanja,
      this.isChecked});

  factory KorisnikIzabranaKnjiga.fromJson(Map<String, dynamic> json) =>
      _$KorisnikIzabranaKnjigaFromJson(json);

  Map<String, dynamic> toJson() => _$KorisnikIzabranaKnjigaToJson(this);
}
